src/apps/filters/h3ToComponents.c

```diff
@@ -100,33 +83,64 @@ void doCell(H3Index h, int verboseMode) {
                    resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
         }
         printf("╚════════════╝\n\n");
+    } else {
+        if (h3Mode == H3_HEXAGON_MODE) {
+            printf("%d:%d:%d:", h3Mode, h3Res, h3BaseCell);
+            for (int i = 1; i <= h3Res; i++) {
+                printf("%c", resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
+            }
+            printf("\n");
+        } else if (h3Mode == H3_UNIEDGE_MODE) {
+            printf("%d:%d:%d:%d:", h3Mode, H3_GET_RESERVED_BITS(h), h3Res,
+                   h3BaseCell);
+            for (int i = 1; i <= h3Res; i++) {
+                printf("%c", resDigitToChar(H3_GET_INDEX_DIGIT(h, i)));
+            }
+            printf("\n");
+        } else {
+            printf("INVALID INDEX\n");
+        }
     }
 }
 
 int main(int argc, char* argv[]) {
-    // check command line args
-    int verboseMode = 0;
-    if (argc > 1) {
-        if (strncmp("--verbose", argv[1], 9) == 0) {
-            verboseMode = 1;
-        } else {
-            fprintf(stderr, "usage: %s [--verbose]\n", argv[0]);
-            exit(1);
-        }
+    H3Index index = 0;
+
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg verboseArg = {.names = {"-v", "--verbose"},
+                      .helpText = "Verbose output mode."};
+    Arg indexArg = {
+        .names = {"-i", "--index"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "index",
+        .value = &index,
+        .helpText =
+            "Index, or not specified to read indexes from standard in."};
```

Comment:
Nit: `stdin` is actually clearer IMO. I think the long form is "standard input".